downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile.java

```diff
@@ -0,0 +1,413 @@
+                    if (activeUnitSchemaClass.isa(ReactomeJavaConstants.Complex) || activeUnitSchemaClass.isa(ReactomeJavaConstants.Polymer)) {
+                        return false;
+                    }
+                    if (activeUnitSchemaClass.isa(ReactomeJavaConstants.EntitySet) && !onlyEWASMembers((activeUnitInstances).get(0))) {
+                        return false;
+                    }
+                } else {
+                    return false;
+                }
+                return true;
+            } else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // This checks if the protein in question has UniProt as a Reference Database, and if the species has a crossReference. If not, it's information will not be generated for the GOA file.
+    // TODO: Rule check
+    private static boolean validProtein(GKInstance referenceEntityInst, GKInstance speciesInst) throws Exception {
+        if (referenceEntityInst != null && speciesInst != null) {
+            GKInstance referenceDatabaseInst = (GKInstance) referenceEntityInst.getAttributeValue(ReactomeJavaConstants.referenceDatabase);
+            if (referenceDatabaseInst != null && referenceDatabaseInst.getDisplayName().equals("UniProt") && speciesInst.getAttributeValue(ReactomeJavaConstants.crossReference) != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Finds most recent modification date for a GOA line. This is a bit of a moving target since GOA lines can be generated convergently for
+    // each type of GO annotation. Depending on if it is looking at the individual protein or whole reaction level, the date attribute
+    // may not be the most recent. If it is found that the goaLine was generated earlier but that a more recent modification date exists based on the
+    // entity that is currently being checked, then it will just update that date value in the hash associated with the line. (Yes, this is weird).
+    private static Integer assignDateForGOALine(GKInstance entityInst, String goaLine) throws Exception {
+        int instanceDate;
+        Collection<GKInstance> modifiedInstances = entityInst.getAttributeValuesList(ReactomeJavaConstants.modified);
+        if (modifiedInstances.size() > 0) {
+            List<GKInstance> modifiedInstancesList = new ArrayList<>(modifiedInstances);
+            GKInstance mostRecentModifiedInst = modifiedInstancesList.get(modifiedInstancesList.size() - 1);
+            instanceDate = Integer.valueOf(mostRecentModifiedInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+        } else {
+            GKInstance createdInst = (GKInstance) entityInst.getAttributeValue(ReactomeJavaConstants.created);
+            instanceDate = Integer.valueOf(createdInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
```

Comment:
The common logic to parse the dateTime attribute value can be consolidated to a function.  A function returning either the most recent modified instance edit or created instance edit would be good to extract too.

Reply:
I created a `getDate` function but since the most recent modified instance is a single conditional now I left that